src/compiler/checker.ts

```diff
@@ -7610,6 +7610,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         if (getDeclarationModifierFlagsFromSymbol(propertySymbol) & (ModifierFlags.Private | ModifierFlags.Protected) && context.tracker.reportPrivateInBaseOfClassExpression) {
                             context.tracker.reportPrivateInBaseOfClassExpression(unescapeLeadingUnderscores(propertySymbol.escapedName));
                         }
+                        // Skip ECMAScript private identifier fields (e.g., #field) in class expression type literals

```

Comment:
Nope. This should be an error here, like a `private`. This is also much more verbose than needed. Try this instead:

```suggestion
                        if (isPrivateIdentifierSymbol(propertySymbol) && context.tracker.reportPrivateInBaseOfClassExpression) {
                            context.tracker.reportPrivateInBaseOfClassExpression(idText((propertySymbol.valueDeclaration! as NamedDeclaration).name! as PrivateIdentifier));
                        }
```